Assets/Scripts/SS3D/Systems/Tile/PlacedObjects/PlacedItemObject.cs

```diff
@@ -0,0 +1,102 @@
+using FishNet;
+using FishNet.Object;
+using SS3D.Core;
+using SS3D.Logging;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace SS3D.Systems.Tile
+{
+    public class PlacedItemObject : NetworkBehaviour
+    {
+        /// <summary>
+        /// Creates a new PlacedItemObject from a prefab at a given position and rotation. Uses NetworkServer.Spawn() if a server is running.
+        /// </summary>
+        /// <param name="worldPosition"></param>
+        /// <param name="origin"></param>
+        /// <param name="rotation"></param>
+        /// <param name="itemSo"></param>
+        /// <returns></returns>
+        public static PlacedItemObject Create(Vector3 worldPosition, Quaternion rotation, ItemObjectSo itemSo)
+        {
+            GameObject placedGameObject = Instantiate(itemSo.prefab);
+            placedGameObject.transform.SetPositionAndRotation(worldPosition, rotation);
+
+            PlacedItemObject placedObject = placedGameObject.GetComponent<PlacedItemObject>();
+            if (placedObject == null)
```

Comment:
In what case can this happen ? Bad set up in editor ? I'd rather avoid any AddComponent if this is the only case it can happen. If it is the case, put a PunPun Panic message instead and return.

Reply:
This component is almost always added at runtime. Not doing so would require every tile and item in the tilemap to have this component added beforehand.

Reply:
Would it be possible to add it automatically but not at runtime ? Maybe using a custom inspector for a particular script ? If it's too tedious or you can't find anything simple, simply add a comment containing something along the lines of your answer to this conversation, as well as a statement that it'd be better if the component did not need to be added and we need to find a better solution.

Reply:
There isn't really a script that is always part of a tilemap item or tile. And we can't add a RequireComponent to the ScriptableObjects that defines which prefabs are part of the tilemap. 

The only way that I see feasible as an alternative is to make an script that runs in the editor that loops through every prefab linked in either a ItemSO or TileSO on Revalidate(), and that will add the component if it is missing. 

Not sure if making that outweighs the costs of just adding it runtime now.